config/routes.rb

```diff
@@ -1,62 +1,64 @@
+      put '/checkout/registration' => 'checkout#update_registration', :as => :update_checkout_registration
+      resource :account, controller: 'users'
+    end
 
-  if Spree.respond_to?(:admin_path) && Spree::Core::Engine.backend_available?
-    namespace :admin, path: Spree.admin_path do
-      devise_for :spree_user,
-                class_name: Spree.user_class.to_s,
-                controllers: { sessions: 'spree/admin/user_sessions',
-                                  passwords: 'spree/admin/user_passwords' },
-                skip: [:unlocks, :omniauth_callbacks, :registrations],
-                path_names: { sign_out: 'logout' },
-                path_prefix: :user
-      devise_scope :spree_user do
-        get '/authorization_failure', to: 'user_sessions#authorization_failure', as: :unauthorized
-        get '/login' => 'user_sessions#new', :as => :login
-        post '/login' => 'user_sessions#create', :as => :create_new_session
-        get '/logout' => 'user_sessions#destroy', :as => :logout
-        get '/password/recover' => 'user_passwords#new', :as => :recover_password
-        post '/password/recover' => 'user_passwords#create', :as => :reset_password
-        get '/password/change' => 'user_passwords#edit', :as => :edit_password
-        put '/password/change' => 'user_passwords#update', :as => :update_password
+    if Spree.respond_to?(:admin_path) && Spree::Core::Engine.backend_available?
+      namespace :admin, path: Spree.admin_path do
+        devise_for :spree_user,
+                  class_name: Spree.user_class.to_s,
+                  controllers: { sessions: 'spree/admin/user_sessions',
+                                    passwords: 'spree/admin/user_passwords' },
+                  skip: [:unlocks, :omniauth_callbacks, :registrations],
+                  path_names: { sign_out: 'logout' },
+                  path_prefix: :user
+        devise_scope :spree_user do
+          get '/authorization_failure', to: 'user_sessions#authorization_failure', as: :unauthorized
+          get '/login' => 'user_sessions#new', :as => :login
+          post '/login' => 'user_sessions#create', :as => :create_new_session
+          get '/logout' => 'user_sessions#destroy', :as => :logout
+          get '/password/recover' => 'user_passwords#new', :as => :recover_password
+          post '/password/recover' => 'user_passwords#create', :as => :reset_password
+          get '/password/change' => 'user_passwords#edit', :as => :edit_password
+          put '/password/change' => 'user_passwords#update', :as => :update_password
+        end
       end
     end
-  end
 
-  namespace :api, defaults: { format: 'json' } do
-    namespace :v2 do
-      namespace :storefront do
-        resource :account, controller: :account, only: %i[show create update]
-        resources :account_confirmations, only: %i[show]
-        resources :passwords, controller: :passwords, only: %i[create update]
+    namespace :api, defaults: { format: 'json' } do
```

Comment:
The API in Spree is not locale-scoped, so these routes should go outside of the `scope :locale` thing

Reply:
For them, locale is always passed as a query param

Reply:
Oh, that makes sense. I've fixed it.